### PR TITLE
bug/Checkbox declaration file inconsistency

### LINKED ISSUE #126: Checkbox declaration file inconsistency
v2.2.0

Hi, I'm seeing a weird inconsistency in the declaration files for `Checkboxes` that makes using `inputRef` prop on `Checkboxes.Box` difficult to use.

In `dist/components/checkboxes/components/Box.d.ts` the definition is as expected

```
declare type BoxProps = Omit<HTMLProps<HTMLInputElement>, 'label'> & {
    labelProps?: LabelProps;
    hint?: string;
    hintProps?: HintProps;
    conditional?: ReactNode;
    forceShowConditional?: boolean;
    conditionalWrapperProps?: HTMLProps<HTMLDivElement>;
    inputRef?: MutableRefObject<HTMLInputElement | null>;
}
```

However the nested definition in `dist/components/checkboxes/Checkboxes.d.ts` has

```
static Box: React.FC<Omit<React.HTMLProps<HTMLInputElement>, "label"> & {
    labelProps?: import("../label/Label").LabelProps;
    hint?: string;
    hintProps?: import("../hint/Hint").HintProps;
    conditional?: React.ReactNode;
    forceShowConditional?: boolean;
    conditionalWrapperProps?: React.HTMLProps<HTMLDivElement>;
    inputRef?: React.MutableRefObject<HTMLInputElement>;
}>;
```

Specifically looking at the inputRef prop:

`inputRef?: MutableRefObject<HTMLInputElement | null>;`
`inputRef?: React.MutableRefObject<HTMLInputElement>;`

I'm not sure what would cause this discrepancy and I'm not entirely sure how to initialise a `useRef` in a way that would satisfy the latter?

Thanks

### PR DESCRIPTION
**Pull request by KaiSpencer**
- Adds strict null checks to fix #126

### CODE REVIEW

File: src/util/HeadingLevel.tsx
Comment (Tomdango) [inline src/util/HeadingLevel.tsx:1]: I'd do a slightly different fix...
You should never hit that `default` case - unless you ignore the type errors and enter something odd into the `headingLevel` prop. I'd just go for keeping it optional, but changing the props of the component to:
```js
({ headingLevel = 'h3', ...rest })
```
Which should clear the type issue.

I'd also be inclined to either raise an error or a warning in the console if we enter that `default` block.

File: tsconfig.json [hunk 1]
```diff
@@ -12,7 +12,8 @@
         "types": ["react", "jest", "node"],
         "allowSyntheticDefaultImports": true,
         "esModuleInterop": true,
-        "skipLibCheck": true
+        "skipLibCheck": true,
+        "strictNullChecks": true
```
Comment (Tomdango) [inline tsconfig.json:16]: :+1: For this

**Review by Tomdango — DISMISSED** [review]
Approved, pending the type fix 👍

File: src/util/HeadingLevel.tsx
Comment (KaiSpencer) [reply src/util/HeadingLevel.tsx:1]: Are you happy with an error thrown? Could be a breaking change if somebody is passing in dodgy data and getting the defaultProps currently.
Might be better for a warning in minor and then breaking in with the v5 branch?

File: src/util/HeadingLevel.tsx [hunk 2]
```diff
@@ -32,10 +32,10 @@ const HeadingLevel: React.FC<HeadingLevelProps> = ({ headingLevel, ...rest }) =>
       return <h5 {...rest} />;
     case 'h6':
       return <h6 {...rest} />;
+    default:
+      console.error(`HeadingLevel: Invalid headingLevel prop.`);
```
Comment (Tomdango) [inline src/util/HeadingLevel.tsx:36]: I'd also print the headingLevel prop provided: `HeadingLevel: Invalid headingLevel prop: ${headingLevel}` - makes it easier for a dev to pinpoint

**Review by Tomdango — APPROVED** [review]

**Pull request merged**